baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/ConsistentOverrides.java

```diff
@@ -0,0 +1,122 @@
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.immutables.value.Value.Immutable;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "ConsistentOverrides",
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        providesFix = BugPattern.ProvidesFix.REQUIRES_HUMAN_ATTENTION,
+        severity = SeverityLevel.ERROR,
+        summary = "Method overrides should have variable names consistent with the super-method when there "
+                + "are multiple parameters with the same type to avoid incorrectly binding values to variables.")
+public final class ConsistentOverrides extends BugChecker implements MethodTreeMatcher {
+    private static final Pattern UNKNOWN_ARG_NAME = Pattern.compile("^args\\d+$");
+
+    @Override
+    public Description matchMethod(MethodTree tree, VisitorState state) {
+        Symbol sym = ASTHelpers.getSymbol(tree);
+        if (!(sym instanceof MethodSymbol) || sym.isStatic()) {
+            return Description.NO_MATCH;
+        }
+
+        MethodSymbol methodSymbol = (MethodSymbol) sym;
+        List<? extends VariableTree> paramTrees = tree.getParameters();
+        getNonParameterizedSuperMethod(methodSymbol, state.getTypes())
+                .filter(ConsistentOverrides::hasMeaningfulArgNames)
+                .ifPresent(superMethod -> {
+                    Map<Type, List<ParamEntry>> superParamsByType = IntStream.range(0, paramTrees.size())
+                            .mapToObj(i -> ParamEntry.of(superMethod.params().get(i), i))
+                            .collect(Collectors.groupingBy(ParamEntry::type));
+
+                    superParamsByType.values().forEach(params -> {
+                        if (params.size() <= 1) {
+                            return;
+                        }
+
+                        for (ParamEntry expectedParam : params) {
+                            int index = expectedParam.index();
+                            String param = methodSymbol.params.get(index).name.toString();
+                            if (param.equals(expectedParam.name())) {
+                                continue;
+                            }
+
+                            state.reportMatch(buildDescription(tree)
+                                    .addFix(SuggestedFixes.renameVariable(
+                                            paramTrees.get(index), expectedParam.name(), state))
+                                    .build());
+                        }
+                    });
+                });
+
+        return Description.NO_MATCH;
+    }
+
+    private static boolean hasMeaningfulArgNames(MethodSymbol methodSymbol) {
+        return !methodSymbol.params().stream()
+                .allMatch(symbol -> UNKNOWN_ARG_NAME.matcher(symbol.name).matches());
```

Comment:
Should we consider single-character names non-meaningful? Otherwise we'll fight checkstyle.